cli/app/commands/conflict/conflict.py

```diff
@@ -0,0 +1,391 @@
+                minor = parsed.minor
+                return f">={base_version}, <{major}.{minor + 1}.0"
+            except Exception:
+                return f">={base_version}"
+
+        # Handle ranges with commas: 1.20.0,2.0.0 -> >=1.20.0, <2.0.0
+        if "," in requirement and not any(op in requirement for op in [">=", "<=", ">", "<", "==", "!="]):
+            parts = [p.strip() for p in requirement.split(",")]
+            if len(parts) == 2:
+                return f">={parts[0]}, <{parts[1]}"
+
+        # Handle x.x.x format ranges: 1.20.* -> >=1.20.0, <1.21.0
+        if requirement.endswith(".*") or requirement.endswith(".x"):
+            base_version = requirement.replace(".*", ".0").replace(".x", ".0")
+            try:
+                parsed = version.parse(base_version)
+                major = parsed.major
+                minor = parsed.minor
+                return f">={base_version}, <{major}.{minor + 1}.0"
+            except Exception:
+                return f">={base_version}"
+
+        # Return as-is if it already contains operators
+        return requirement
+
+
+class ConfigLoader:
+    """Handles loading and parsing of configuration files."""
+    
+    def __init__(self, logger: LoggerProtocol):
+        self.logger = logger
+
+    def load_config(self, config_path: str) -> Dict[str, Any]:
+        """Load configuration from YAML file."""
+        self.logger.debug(conflict_loading_config.format(path=config_path))
+
+        if not os.path.exists(config_path):
+            raise FileNotFoundError(conflict_config_not_found.format(path=config_path))
+
+        try:
+            with open(config_path, "r") as f:
+                config = yaml.safe_load(f)
+
+            self.logger.debug(conflict_config_loaded)
+            return config
+        except yaml.YAMLError as e:
+            raise ValueError(conflict_invalid_config.format(error=str(e)))
+        except Exception as e:
+            raise Exception(conflict_invalid_config.format(error=str(e)))
+
+
+class ToolVersionChecker:
+    """Handles version checking for different tools."""
+    
+    # Common version commands for different tools
+    VERSION_COMMANDS = {
+        "docker": ["docker", "--version"],
+        "docker-compose": ["docker-compose", "--version"],
+        "go": ["go", "version"],
+        "node": ["node", "--version"],
+        "npm": ["npm", "--version"],
+        "yarn": ["yarn", "--version"],
+        "python": ["python", "--version"],
+        "python3": ["python3", "--version"],
+        "pip": ["pip", "--version"],
+        "pip3": ["pip3", "--version"],
+        "git": ["git", "--version"],
+        "curl": ["curl", "--version"],
+        "ssh": ["ssh", "-V"],
+        "caddy": ["caddy", "version"],
+        "postgresql": ["psql", "--version"],
+        "redis": ["redis-server", "--version"],
+        "air": ["air", "-v"],
+    }
+    
+    # Tool name mappings for command execution
+    TOOL_MAPPING = {
```

Comment:
keeping in config yaml file would help

Reply:
Again, better to keep it as part of constants? What do you think?

My thought proces is environment variables or specific values in the config, but anything that goes beyond that, like be part of executional logic  like cmds or data mappings etc could go to constants.py?

suggesting this approach to avoid bloating `config.prod.yml`

incase going ahead someone wants to create a custom config.yml, should have too many things i feel.. easy to understand and interpret by looking into it.. moving commands and other things might be lil overwhelming i feel.